pkg/resource/detector/detector.go

```diff
@@ -0,0 +1,101 @@
+package detector
+
+import (
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"time"
+)
+
+// Detector represents a procedure that runs in the background, periodically auto-detecting features
+type Detector struct {
+	dc                  discovery.DiscoveryInterface
+	ticker              *time.Ticker
+	SubscriptionChannel chan schema.GroupVersionKind
```

Comment:
@philbrookes It doesn't look like the channel is used anymore with your trigger changes.

Reply:
good point. I'll remove it